plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt

```diff
@@ -89,10 +86,11 @@ object StmtConversionVisitor : FirVisitor<ExpEmbedding, StmtConversionContext<Re
 
     override fun visitWhenExpression(whenExpression: FirWhenExpression, data: StmtConversionContext<ResultTrackingContext>): ExpEmbedding {
         val subj = whenExpression.subject?.let { data.convertAndStore(it) }
-        val ctx = if (whenExpression.usedAsExpression) {
-            data.withResult(data.embedType(whenExpression))
+        val type = data.embedType(whenExpression)
+        val ctx = if (type != NothingTypeEmbedding) {
```

Comment:
Would also doing this with `UnitTypeEmbedding` make sense?

Reply:
I was thinking about it but probably this will not work
```
    val x = when {
        else -> println(1)
    }
```

Reply:
I think we still evaluate the branch, though?  So the side effects should still happen, we just don't have a result (and so result is the unit value).